src/Microsoft.AspNetCore.Blazor.Server/LiveReloadingContext.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Blazor.Server
+{
+    internal class LiveReloadingContext
+    {
+        // Keep in sync with $(BlazorBuildCompletedSignalPath) in Blazor.MonoRuntime.props
+        private const string BlazorBuildCompletedSignalFile = "__blazorBuildCompleted";
+
+        // If some external automated process is writing multiple files to wwwroot,
+        // you probably want to wait until they've all been written before reloading.
+        // Pausing by 500 milliseconds is a crude effort - we might need a different
+        // mechanism (e.g., waiting until writes have stopped by 500ms).
+        private const int WebRootUpdateDelayMilliseconds = 500;
```

Comment:
We could also consider using files as a notification. Write a file with a particular name as the first change and then delete it when the build is complete.

Reply:
That's (almost) exactly what we do :) https://github.com/aspnet/Blazor/pull/193/files#diff-017c9b5955fe407916ebe35f7ee07012R63

Only difference is we both write and delete it at once, when the build is complete. We don't need to notify the client that a build is beginning (at least we don't have any need for that yet).